src/App.tsx

```diff
@@ -8,53 +8,175 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import {
+  addComments, deleteComment, getComments, getPosts, getUsers,
+} from './api/posts';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isError, setIsError] = useState(false);
+  const [isSelected, setIsSelected] = useState(false);
+
+  async function fetchUsers() {
+    const usersList = await getUsers();
+
+    setUsers(usersList);
+  }
+
+  async function fetchPosts() {
+    setIsLoading(true);
+    try {
+      const postsList = await getPosts();
+
+      setPosts(postsList);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  async function fetchComments() {
+    setIsLoading(true);
+    try {
+      const commentsList = await getComments();
+
+      setComments(commentsList);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  useEffect(() => {
+    fetchUsers();
+  }, []);
+
+  const selectUser = (user: User) => {
+    setSelectedUser(user);
+  };
+
+  const selectPost = (post: Post) => {
+    setSelectedPost(post);
+  };
+
+  useEffect(() => {
+    fetchPosts();
+  }, [selectedUser]);
+
+  useEffect(() => {
+    fetchComments();
+  }, [selectedPost]);
+
+  const filteredPosts = posts
+    .filter(post => selectedUser?.id === post.userId) || [];
+  const filteredComments = comments
```

Comment:
Simplify the code by removing '|| []' since Array.prototype.filter always returns an array.